docs/source/platform/schema-validation.mdx

```diff
@@ -12,18 +16,24 @@ Apollo provides a tool to protect for exactly this scenario called **schema vali
 ## How it works
 
 Schema validation is run through the Apollo CLI by executing the `apollo service:check` command. Apollo will generate a diff between your local schema and your most recently registered schema, then validate that the changes are safe by checking if any queries actively running against your graph will be affected.
+If your graph is composed using federation, you can also use the `apollo service:check` command to validate individual service changes by adding the `--serviceName` flag. However, unlike running service check on a non-federated service, Engine first needs to run composition on the proposed partial schema with the current list of implementing services and their partial schemas. Composition must pass before validation can continue.
+
+But wait... What exactly is composition? 🤔 In order to have different implementing services contribute to the same graph, there needs to be a way to combine each implementing service's partial schemas into one large data graph. Engine's composition algorithm does exactly this.
 
-Here's how it works:
+Here's how schema validation works for federated and non-federated services:
 
-1. You run `apollo service:check` locally or in CI. The proposed schema is sent to Engine's schema registry.
-1. Engine creates a diff between the local schema and the most recently published schema in the registry.
+1. You run `apollo service:check` locally or in CI. For federated services run the same command but add `--serviceName=ExampleService --endpoint=http://loocalhost:4000/graphql` to the end. The proposed schema/partial schema (if using federation) is sent to Engine's schema registry.
+1. If you're using federation, Engine's schema registry will attempt to compose the partial schema with the current schema. If composition fails then we do not continue with validation and exit early.
+1. Engine creates a diff between the local schema and the most recently published schema in the registry. For federation this step happens after composition passes.
```

Comment:
```suggestion
1. Engine creates a diff between the local schema and the most recently published schema in the registry.
```

Do we need the last sentence?

Reply:
meh no we don't need it. i think it's clear enough from the step above